functions-and-scope.js

```diff
@@ -39,46 +57,77 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
+// Declareer een variabele die de optelling wordt van alle cijfers.
+// Loop door de array en tel de cijfers op
+// Na de loop deel je het totaal van de cijfers door de lengte van de begin array
+// retourneer de uitkomst
+// console log
 
 /* 2b: Omschrijven tot een herbruikbare functie */
 // Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
-
 // ---- Verwachte uitkomsten:
 // averageGrade(grades) geeft 6.642857142857143
 // averageGrade([6, 4, 5]) geeft xxxx
 // averageGrade([8, 9, 4, 6, 10]) geeft xxxx
 
+// ** ANTWOORD 2b **
+function averageGrade(value){
+    let cumulativeGrades = 0;
+    for (let i = 0; i < value.length; i++){
+        cumulativeGrades += value[i];
+        }
+    return Math.round(cumulativeGrades / value.length * 100) / 100;
+    }
+
+console.log(averageGrade(grades));
+console.log(averageGrade([6, 4, 5]));
+console.log(averageGrade([8, 9, 4, 6, 10]));
 
 /* 2c: Afronden op twee decimalen */
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
-
-
+// ** ANTWOORD 2c **
+// boven in de return statement aangepast
 
 /* Bonusopdracht: hoogste cijfer */
-
 /* 3a: Script schrijven  */
 // Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe kan ik iedere waarde van de array langsgaan?
 // * Op welke conditie moet ik checken?
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
 // Log het antwoord in de terminal.
-
 // ---- Verwachte uitkomst: 9
 
+// ** ANTWOORD 3a **
+// declareer een variabele wat het hoogste cijfer zal bevatten
+// itereer door de array
+// bij iedere index afvragen of de waarde van de array hoger is dan de variabelen die we eerst gedeclareerd hebben.
+// indien de waarde hoger is overschrijf de waarde van de variabelen.
+// retourneer de variabele
 
 /* 3b: Omschrijven tot een herbruikbare functie */
 // Schrijf een functie genaamd highestGrade, die een array van cijfers verwacht (zoals grades) en het hoogste cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 3a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
-// ---- Verwachte uitkomsten:
-// highestGrade(grades) geeft 9
-// highestGrade([6, 4, 5]) geeft 6
-// highestGrade([8, 9, 4, 6, 10]) geeft 10
+// ** ANTWOORD 3b **
+function highestGrade(value){
+    let highestGrade = 0;
```

Comment:
Je kan hier beter niet 0 maar de eerste waarde uit de array pakken. Het zal hier altijd goed gaan, omdat in de praktijk cijfers niet lager dan 0 kunnen zijn. Maar stel je voor dat je een array aanlevert van [-1, -2, -5, -3], dan krijg je 0 terug wat dus niet klopt.